File: src/main/scala/dev/tlang/tlang/ast/common/value/MultiValue.scala
package dev.tlang.tlang.ast.common.value

import dev.tlang.tlang.ast.common.call.ComplexValueStatement
import dev.tlang.tlang.ast.common.{ObjType, ValueType}
import dev.tlang.tlang.tmpl.AstContext
import tlang.core.{Type, Value}
import tlang.internal.{ClassType, ContextContent}

case class MultiValue(context: Option[ContextContent], values: List[Value]) extends ComplexValueStatement[MultiValue] with AstContext {

  override def getType: Type = MultiValue.getType


  override def getContext: Option[ContextContent] = context
}

object MultiValue extends TLangType {
  override def getType: Type = ClassType.of(classOf[MultiValue])

  override def getValueType: ValueType = ObjType(None, Some("TLang"), getType)
}

File: src/main/java/tlang/core/MultiValue.java
package tlang.core;

import tlang.internal.ClassType;

public class MultiValue implements Value {

    private final Value[] values;

    public MultiValue(Value... values) {
        this.values = values;
    }

    public Value[] getValues() {
        return values;
    }

    @Override
    public MultiValue getValue() {
        return this;
    }

    @Override
    public Type getType() {
        return ClassType.of(MultiValue.class);
    }
}
